tests/run/constant_folding.py

```diff
@@ -305,9 +305,10 @@ def for_in_empty_nested_listcomp():
 
 @cython.test_fail_if_path_exists(
     "//ForInStatNode//ForInStatNode",
+    "//ForFromStatNode//ForFromStatNode",
```

Comment:
```suggestion
@cython.test_fail_if_path_exists(
    "//ForInStatNode",
    "//ForFromStatNode//ForFromStatNode",
```

Reply:
Don't agree with this change. The test is about whether constants are folded, not how it's optimized. The failure is that there are two loops (whether they're `ForInStatNode` or `ForFromStatNode` loops).

Reply:
Ok, but there also shouldn't be a mix of the two types. I think that's what the assertion tried to cover.
```suggestion
@cython.test_fail_if_path_exists(
    "//LoopNode//LoopNode",
```

Reply:
Unfortunately the test_*_path_exists decorators don't cover inheritance (and making them account for inheritance breaks a lot, e.g. assert PythonCapiCallNode, fail if SimpleCallNode). Since I'm no longer applying this to tuples and lists I think the easiest thing is to revert to what it was before.